components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -268,24 +364,38 @@ void Nudging::run_impl (const double dt)
     });
 
     // Vertical Interpolation onto atmosphere state pressure levels
+    // Note that we are going from ext to hxt here
     if (m_src_pres_type == TIME_DEPENDENT_3D_PROFILE) {
       perform_vertical_interpolation<Real,1,2>(p_mid_ext_p,
                                                p_mid_v,
                                                ext_state_view,
-                                               int_state_view,
+                                               hxt_state_view,
                                                int_mask_view,
                                                m_num_src_levs,
                                                m_num_levs);
     } else if (m_src_pres_type == STATIC_1D_VERTICAL_PROFILE) {
       perform_vertical_interpolation<Real,1,2>(p_mid_ext_1d,
                                                p_mid_v,
                                                ext_state_view,
-                                               int_state_view,
+                                               hxt_state_view,
                                                int_mask_view,
                                                m_num_src_levs,
                                                m_num_levs);
     }
 
+    // Refine-remap onto target atmosphere state horiz grid ("int")
+    // Note that we are going from hxt to int here
+    if(m_refine_remap) {
+      // Call the remapper
+      print_field_hyperslab (ext_state_field);
+      print_field_hyperslab (hxt_state_field);
+      print_field_hyperslab (int_state_field);
+      refine_remapper->remap(true);
```

Comment:
You are calling the remap method inside a fol loop, causing all fields to be remapped multiple times. The fields not yet remapped vertically will likely still have invalid values, no?